src/multibody/liegroup/cartesian-product-variant.hxx

```diff
@@ -363,6 +363,25 @@ normalize_impl (const Eigen::MatrixBase<Config_t>& qout) const
   }
 }
 
+template<typename _Scalar, int _Options, template<typename,int> class LieGroupCollectionTpl>
+template <class Config_t>
+bool CartesianProductOperationVariantTpl<_Scalar,_Options,LieGroupCollectionTpl>::
+isNormalized_impl (const Eigen::MatrixBase<Config_t>& qin,
+                   const Scalar prec) const
+{
+  Index id_q = 0;
+  for(size_t k = 0; k < liegroups.size(); ++k)
+  {
+    const Index & nq = lg_nqs[k];
```

Comment:
Don't use ref here.

Reply:
This was largely inspired from the other algorithms in the file, which all use a reference at this point. I agree it is useless. I can remote it from here, no problem, but then I recommend to do the same in all other algorithms

Reply:
Yes, I will do it for the rest in Pinocchio 3.x